bin/data_frame_analyzer/Main.cc

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+//! \brief
+//! Applies a range of ML analyses on a data frame.
+//!
+//! DESCRIPTION:\n
+//! Expects to be streamed CSV or length encoded data on STDIN or a named pipe,
```

Comment:
I agree it would be nice to have the option to accept CSV.  That would facilitate easier use from the command line, e.g. to find memory bugs with `valgrind`.  However, at the moment the program cannot accept CSV.